sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageController.java

```diff
@@ -693,118 +655,79 @@ private FacilityReferenceDto getLabReference(LabMessageDto labMessageDto) {
+		CommitDiscardWrapperComponent<SampleCreateForm> sampleCreateComponent = sampleController.getSampleCreateComponent(sample, disease, () -> {
+		});
 
-		CheckBox viaLimsCheckbox = sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.VIA_LIMS);
-		if (viaLimsCheckbox != null) {
-			viaLimsCheckbox.setValue(Boolean.TRUE);
-			viaLimsCheckbox.setEnabled(false);
-		}
-		// TODO currently just the first testReport is picked here. That must be temporary. Will be fixed with #5899
-		List<TestReportDto> testReportDtos = FacadeProvider.getTestReportFacade().getAllByLabMessage(labMessageDto.toReference());
-		if (!testReportDtos.isEmpty()) {
-			TestReportDto testReportDto = testReportDtos.get(0);
-			((TextField) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.EXTERNAL_ID)).setValue(testReportDto.getExternalId());
-			((TextField) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.EXTERNAL_ORDER_ID))
-				.setValue(testReportDto.getExternalOrderId());
-			((ComboBox) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.TEST_RESULT)).setValue(testReportDto.getTestResult());
-			((ComboBox) sampleCreateComponent.getWrappedComponent().getField(SampleDto.PATHOGEN_TEST_RESULT)).setValue(testReportDto.getTestResult());
-			((ComboBox) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.TEST_TYPE)).setValue(testReportDto.getTestType());
-			((ComboBox) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.TESTED_DISEASE))
-				.setValue(labMessageDto.getTestedDisease());
-			((NullableOptionGroup) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.TEST_RESULT_VERIFIED))
-				.setValue(testReportDto.isTestResultVerified());
-			((DateTimeField) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.TEST_DATE_TIME))
-				.setValue(testReportDto.getTestDateTime());
-			if (testReportDto.getTypingId() != null) {
-				Field typingIdField = sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.TYPING_ID);
-				typingIdField.setValue(testReportDto.getTypingId());
-				typingIdField.setVisible(true);
-			}
-		}
-		if (FacadeProvider.getConfigFacade().isConfiguredCountry(CountryHelper.COUNTRY_CODE_GERMANY)) {
-			((DateField) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.REPORT_DATE))
-				.setValue(labMessageDto.getMessageDateTime());
+		sampleCreateComponent.addDoneListener(() -> finishProcessingLabMessage(labMessageDto, sample));
+
+		// add pathogen test create components
+		//********************
+		List<PathogenTestDto> pathogenTests = buildPathogenTests(sample, labMessageDto);
+		int caseSampleCount = FacadeProvider.getSampleFacade().caseSampleCountOf(sample);
+		// the first pathogenTestCreateComponent must not be removable. It is expected to exist because the buildPathogenTests shall always return at least one.
+		PathogenTestForm pathogenTestCreateComponent =
```

Comment:
It would be better to don't let to delete the last remaining pathogen test then forcing the first one to be not deletable. What if the user would like to keep the third one only?

Reply:
With the current approach, the user would need to copy the values from the third pathogen test to the first one. There are not too many fields... I'm open to both approaches, but I think the one proposed by you also has ownsides.
- it is a little more complex and puts a little more load on the UI. 
- the remove pathogen test button would show for all attached pathogen test forms (when there are more than one). That may suggest to some users that all pathogen tests are removable. 

Considering this, do you still prefer your proposal?

Reply:
Yes, I prefer my proposal because it's a better UX

Reply:
The lifecycle scenarios of the new profiles allow lab messages to only correct data. Processing this should not require a pathogen test to exist for a related sample. 
The restraint to always keep at least one pathogen test was therefore removed by me completely.